# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: gfx/layers/CanvasDrawEventRecorder.cpp
# Commit: b74ed8e9d95a
# Full Hash: b74ed8e9d95a8907add8dd764ccc89be1c7ab99d
# Author: Andrew Osmond <aosmond@mozilla.com>
# Date: 2023-12-13 09:27:26
# Description:
#   Bug 1869658 - Fix usage of CanvasBuffer on failure paths. r=gfx-reviewers,lsalzman
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D196232
# ==============================================================================

diff -r 3f82f965831a -r b74ed8e9d95a gfx/layers/CanvasDrawEventRecorder.cpp
--- a/gfx/layers/CanvasDrawEventRecorder.cpp	Wed Dec 13 03:45:15 2023 +0000
+++ b/gfx/layers/CanvasDrawEventRecorder.cpp	Wed Dec 13 03:45:16 2023 +0000
@@ -210,13 +210,13 @@
   auto newBuffer = CreateAndMapShmem(bufferSize);
   if (NS_WARN_IF(newBuffer.isNothing())) {
     mHeader->writerState = State::Failed;
-    mCurrentBuffer = CanvasBuffer(nullptr);
+    mCurrentBuffer = CanvasBuffer();
     return mCurrentBuffer;
   }
 
   if (!mHelpers->AddBuffer(std::move(newBuffer->handle), bufferSize)) {
     mHeader->writerState = State::Failed;
-    mCurrentBuffer = CanvasBuffer(nullptr);
+    mCurrentBuffer = CanvasBuffer();
     return mCurrentBuffer;
   }
 